lib/make/makefile.deeparg

```diff
@@ -0,0 +1,217 @@
+# this file.
+DIAMOND_ALIGN_OPTS =  
+
+# Output format of the DIAMOND results. The default value refers to a tabular file.
+# For more format options see https://github.com/bbuchfink/diamond.
+OUTFMT= 6
+
+# IMPORTANT: how to download the deeparg database
+# 1) $ git clone https://bitbucket.org/gusphdproj/deeparg-ss.git
+# then, go to the "database" directory, choose the version (v1 or v2)
+# and link the file "features.fasta" into the DIAMOND work directory
+# 2) install deeparg, e.g. via conda
+# $ conda create -n deeparg_env python=2.7.18
+# $ source activate deeparg_env
+# $ pip install deeparg==1.0.2
+# $ deeparg download_data -o /path/to/local/directory/
+# then look for the "features.fasta" file as above.
+# More about the deeparg installation on bitbucket:
+# https://bitbucket.org/gusphdproj/deeparg-ss/src/master/
+# More about the database versions in this thread:
+# https://bench.cs.vt.edu/argminer/#/forum/selected_question;id=1582104387832
+
+# Download HTTPS URL for the deeparg.py script, necessary for the downstream analyses
+HTTPS = https://gist.github.com/5f3b236bd36acf7a5a7fcf3cff99a56d.git 
+# IMPORTANT: deeparg.py needs to be placed inside the directory that has the DIAMOND output
+# so run mirror_deeparg.py into the directory that contains .argdb.sort.best.m8 files
+# N.B.: the python script "deeparg.py" requires python 3.6*-3.8*
+
+# *** Internal ***
+
+# MAKECALL_DIAMOND_DEEPARG is a macro that defines what will be output to the .makecall
+# file, the file that records versions, file stamps, parameters etc.
+#
+# *Don't redefine!*
+MAKECALL_VERSION     = echo "`date +"%Y-%m-%d %H:%M:%S"`: $@ was made with `diamond --version`" > $@.makecall
+MAKECALL_PARAMS      = echo "	Called with parameters: $(DIAMOND_ALIGN_OPTS)" >> $@.makecall
+MAKECALL_INFILES     = echo "	Input files: $^ (`ls -lL $^|tr '\n' ','`)" >> $@.makecall
+MAKECALL_DB_DEEPARG      = echo "	Database: $(DIAMOND_DB_PATH)/$(DIAMOND_DB) (`ls -lL $(DIAMOND_DB_PATH)/*.dmnd`)" >> $@.makecall
+MAKECALL_DIAMOND_DEEPARG = $(MAKECALL_VERSION); $(MAKECALL_PARAMS); $(MAKECALL_DB_DEEPARG); $(MAKECALL_INFILES)
+MAKECALL_MAKEDB      = $(MAKECALL_VERSION); $(MAKECALL_INFILES)
+
+# *** misc targets ***
+
+# compress all .fastq files in the work directory.
+compress_all_fastqs: $(subst .fastq,.fastq.gz, $(wildcard *.fastq))
+
+# Interleave two fastq files (named %.R1.fastq and %.R2.fastq respectively).
+#
+# The target uses a python script -- interleave_fastq.py -- cloned from this
+# gist:
+#
+#   https://gist.github.com/rpucheq/cf50eeea794529ca6b0e
+#  *** Warning! ***
+# The aforementioned script works for bot .gzip and uncompressed files
+# but it has been written in PYTHON 2!!!
+#
+# Dependencies will automatically be unzipped if not present in unzipped format
+# thanks to the pattern above. Unzipped files will be deleted after
+# interleaving.
+interleave_all_fastq.gzs: $(subst .R1.fastq.gz,.intlv.fastq.gz,$(wildcard *.R1.fastq.gz))
```

Comment:
Also `makefile.misc`. (Also in another file I reviewed above. Define every target only once.)

Reply:
Same as above reply.